### PR TITLE
Create example pure express.js endpoint

### PR DESCRIPTION
**Pull request by khemritolya**
### Summary

This pull request is the first step towards deprecating the Meteor Shim on the backend

This PR:

* Creates an example endpoint to pattern other after
* Sets up a small framework which abstracts away boiler-plate code
* Sets up testing for the newly created endpoints

This is a draft pull request while we decide if this is a good pattern, and if there are any things about the pattern that we want to change.

### Test Plan 

Write tests for the endpoints. See endpoints.tests.ts

### Notes 

TODO: Discuss the merits.

### Breaking Changes 

None. Deprecation will come later.

### CODE REVIEW

File: server/endpoints.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ */
+export function configure(app: express.Application) {
+  const methods = express.Router();
+  methods.use(express.json());
+  app.use(express.json());
+
+  register(app, "getClassesByQuery", getClassesByQuery);
+}
+
+function register<T>(app: express.Application, name: string, endpoint: Endpoint<T>) {
```
Comment (draycfarley) [inline server/endpoints.ts:23]: Maybe this has to do with the shim or using Typescript with Express, but why not use app.use("/v2")? When I've used Express without TS that was sufficient for directing requests to a set of endpoints.

Comment (khemritolya) [reply server/endpoints.ts:23]: I think that this is more explicit about where all the new endpoints are listening. We can consider changing it though.

File: server/endpoints.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+import axios from 'axios';
+import { configure } from "./endpoints";
+import { Classes, Students, Subjects } from './dbDefs';
+
+let mongoServer: MongoMemoryServer;
+
+const testingPort = 17760;
+
+beforeAll(async () => {
```
Comment (AkashAryal) [inline server/endpoints.test.ts:13]: In the future we should probably set up a jest global setup and global teardown so we don't have to repeat this every test file.

Comment (khemritolya) [reply server/endpoints.test.ts:13]: I agree in the long run. However, in the short run, all tests modify the DB, and they do so in different ways. For tests to be deterministic, this is a necessity. Again, maybe in the future we can be smarter about this.

**Review by julian-londono — CHANGES_REQUESTED** [review]
Everything looks good to me!  Could you create an `/endpoints/` directory where we could house back end functions by components in their own folders as discussed during the sync?

E.g. `getClassesByQuery` would go in `SearchBar.ts` because it's used by this component

**Review by julian-londono — APPROVED** [review]

**Pull request merged**